gammapy/maps/hpxnd.py

```diff
@@ -379,6 +379,75 @@ def stack(self, other, weights=None):
             data = data * weights.data
         self.data[..., idx] += data
 
+
+    def smooth(self, width, kernel="gauss"):
+        """Smooth the map.
+
+        Iterates over 2D image planes, processing one at a time.
+
+        Parameters
+        ----------
+        width : `~astropy.units.Quantity`, str or float
+            Smoothing width given as quantity or float. If a float is given it
+            interpreted as smoothing width in pixels. If an (angular) quantity
+            is given it converted to pixels using ``healpy.nside2resol``.
+            It corresponds to the standard deviation in case of a Gaussian kernel,
+            and the radius in case of a disk kernel.
+        kernel : {'gauss', 'disk'}
+            Kernel shape
+
+        Returns
+        -------
+        image : `HpxNDMap`
+            Smoothed image (a copy, the original object is unchanged).
+        """
+        import healpy as hp
+
```

Comment:
I think this currently only works for an allsky map. So maybe raise and error here, if it is a partial HPX map.